cmdmoverename.go

```diff
@@ -141,6 +141,58 @@ func doMoveAfter(script, before, after string) error {
 	return nil
 }
 
+func doMoveBefore(script, before, after string) error {
+	beforePlanPath := before + ".tfplan"
+	beforePlanFile, err := os.Open(beforePlanPath)
+	if err != nil {
+		return fmt.Errorf("opening the terraform BEFORE plan file: %v", err)
+	}
+	defer beforePlanFile.Close()
+
+	upPath := script + "_up.sh"
+	upFile, err := os.Create(upPath)
+	if err != nil {
+		return fmt.Errorf("creating the up file: %v", err)
+	}
+	defer upFile.Close()
+
+	downPath := script + "_down.sh"
+	downFile, err := os.Create(downPath)
+	if err != nil {
+		return fmt.Errorf("creating the down file: %v", err)
+	}
+	defer downFile.Close()
+
+	beforeCreate, beforeDestroy, err := parse(beforePlanFile)
+	if err != nil {
+		return fmt.Errorf("parse BEFORE plan: %v", err)
+	}
+	if beforeCreate.Size() == 0 {
+		return fmt.Errorf("BEFORE plan does not contain resources to create")
+	}
+	if beforeDestroy.Size() > 0 {
+		return fmt.Errorf("BEFORE plan contains resources to destroy: %s",
+			sorted(beforeDestroy.List()))
+	}
+
+	upMatches, downMatches := matchExact(beforeCreate, beforeCreate)
```

Comment:
In the end, it's probably so simple it was hard to see at the beginning, wasn't it? :smile:

Reply:
Exactly :-)

I have also the feeling that the whole code can be simplified more if looked at it from the right distance.